interpreter/chapter.txt

```diff
@@ -1,39 +1,43 @@
-A Python interpreter written in Python
+<!-- American spelling -->
+# A Python Interpreter Written in Python
 
-# Introduction
-This edition of The Architecture of Open Source Applications showcases examples of powerful programs written in 500 lines or less. In this chapter, we'll look at a common, multipurpose tool: a Python interpreter.
+## Introduction
 
-Byterun is a Python interpreter implemented in Python. Through my work on Byterun, I was surprised and delighted to discover that the fundamental structure of the Python interpreter fits easily into the 500-line size restriction. This chapter will walk through the structure of the interpreter and give you enough context to explore further if you're inclined to do so. The goal today is not to explain everything there is to know about interpreters - like so many interesting areas of programming and computer science, you could devote years to developing a deep understanding of the topic.
+Byterun is a Python interpreter implemented in Python. Through my work on Byterun, I was surprised and delighted to discover that the fundamental structure of the Python interpreter fits easily into the 500-line size restriction. This chapter will walk through the structure of the interpreter and give you enough context to explore it further. The goal is not to explain everything there is to know about interpreters --- like so many interesting areas of programming and computer science, you could devote years to developing a deep understanding of the topic.
 
 Byterun was written by Ned Batchelder and myself, building on the work of Paul Swartz. Its structure is similar to the primary implementation of Python, CPython, so understanding Byterun will help you understand interpreters in general and the CPython interpreter in particular. (If you don't know which Python you're using, it's probably CPython.) Despite its short length, Byterun is capable of running most simple Python programs.
 
-## A Python interpreter
-Before we begin, let's narrow down what we mean by "a Python interpreter." The word "interpreter" can be used in a variety of different ways when discussing Python. Sometimes "interpreter" refers to the Python REPL, the interactive prompt you get by typing just "python" from the command line. Sometimes people use "the Python interpreter" more or less interchangeably with "Python" to talk about executing Python code from start to finish. In this chapter, "interpreter" has a more narrow meaning: it's the last step in the process of executing a Python program.
+### A Python Interpreter
 
-Before the interpreter takes over, Python performs three other steps: lexing, parsing, and compiling. Together, these steps transform the programmer's source code from lines of text into structured _code objects_ containing instructions that the interpreter knows how to understand. The interpreter's job is then to take these code objects and follow the instructions.
```

Comment:
Moved this paragraph down below the discussion of interpreters so the paragraph addressing what "interpreter" means comes first.